### ISSUE TITLE
Extend 'in' operator to check for array subset.

### ISSUE THREAD

**Issue by jordansissel** [elastic/logstash#6853]
Currently, the `in` operator does the following for `a in b` example:

* if `b` is a string, it checks if `a` is a substring of `b`
* if `b` is a list, it checks if `a` is an element of `b`

I propose that we extend the `in` operator to have the following additional behavior:

* if `b` is a list, and `a` is a list, then `a in b` should check if `a` is a subset of the elements in `b`

Examples:

* `["a", "b"] in ["a", "b", "c"]` == true
* `["a", "d"] in ["a", "b", "c"]` == false

The same goes for field references. Let's suppose the `[tags]` field is `["web", "frontend", "newyork", "staging"]`, then the following:

* `["web", "frontend"] in [tags]` == true
* `["db", "newyork"] in [tags]` == false

**Comment by justinkambic** [comment]
Are we still looking to create this enhancement? I've added to the relevant spec on my fork, but before I do anything else I wanted to reopen the discussion.

**Comment by AbdulHaseebHussain** [comment]
@justinkambic @jordansissel Is this issue still unresolved?
I could work on it if needed.

**Comment by jordansissel** [comment]
I don't think we currently support this feature, though I haven't checked the test suite. Anyone implementing this is welcome. :)

**Comment by saravanak** [comment]
@AbdulHaseebHussain , are you in progress on this, else I am ready to pitch in, thanks!

**Comment by AbdulHaseebHussain** [comment]
yep sure pitch in, I have not progressed any further.

**Comment by justinkambic** [comment]
@saravanak @AbdulHaseebHussain have either of you progressed on this? I'm in a place where I can take it on, but I don't want to redundantly do anything you've already accomplished!

**Comment by AbdulHaseebHussain** [comment]
@justinkambic I have not progressed any further so it would be great if you could it on.

**Comment by saravanak** [comment]
Me neither, feel free to punch in @justinkambic
